Added ROS2 introduction (#60)
eProsima/vulcanexus: docs/rst/introduction/metapackages.rst

.. _vulcanexus_binary_distribution:

Vulcanexus binary distributions
===============================

*Vulcanexus* provides several binary distributions that install specific packages depending on the developer's needs.
:ref:`vulcanexus_introduction` table summarizes the different *Vulcanexus* package distributions along with the most important included packages.

Vulcanexus Core
---------------

**Vulcanexus Core** is the most fundamental and minimal *Vulcanexus* installation.

> It is composed by the `ROS 2 Base libraries <https://ros.org/reps/rep-2001.html#end-user-entry-points>`_ but with the latest `eProsima Fast DDS <https://fast-dds.docs.eprosima.com/en/latest/>`_ and `Fast CDR <https://www.eprosima.com/images/PDFs/Fast_CDR.pdf>`_ releases and the RMW needed to benefit from the latest features included in those releases.

Reviewer: ```suggestion
It is composed by the `ROS 2 Base libraries <https://ros.org/reps/rep-2001.html#end-user-entry-points>`_ but with the latest `eProsima Fast DDS <https://fast-dds.docs.eprosima.com/en/latest/>`_ and `Fast CDR <https://www.eprosima.com/images/PDFs/Fast_CDR.pdf>`_ releases and the ROS MiddleWare (RMW) needed to benefit from the latest features included in those releases.
```

No additional developer tools are included with this basic installation.

.. note::

    :ref:`tutorials_core_ros2_tutorials` showcase the specific features that *Vulcanexus* implements but are not yet included within the *ROS 2* ecosystem.

.. note::

    The binary package is called following the next convention ``vulcanexus-<version_name>-core`` being the ``version_name`` the corresponding adjective associated with the :ref:`specific release <release_notes>`.
    For example, for Vulcanexus v2, ``version_name`` is ``humble``.
    Consequently, in order to install this package run the following command after setting up the repository as explained in the :ref:`installation guide <linux_binary_installation>`

        .. code-block:: bash

            sudo apt install vulcanexus-humble-core

Vulcanexus Tools
----------------

**Vulcanexus Tools** is focused on helping developers with the introspection and debugging of their system.

> Thus, the `eProsima Fast DDS Monitor <https://fast-dds-monitor.readthedocs.io/en/latest/>`_ and its dependency (`eProsima Fast DDS Statistics Backend <https://fast-dds-statistics-backend.readthedocs.io/en/latest/>`_) are included in order to monitor and diagnose the DDS communication network.

Reviewer: ```suggestion
Thus, the `ROS 2 Monitor <https://fast-dds-monitor.readthedocs.io/en/latest/>`_ and the `ROS 2 Statistics Backend <https://fast-dds-statistics-backend.readthedocs.io/en/latest/>`_ are included in order to monitor and diagnose the ROS 2 communication network.
```

> Also, `eProsima Shapes Demo <https://eprosima-shapes-demo.readthedocs.io/en/latest/index.html>`_ is provided as a simple demo to help testing that the communication is successfully established in the system.

Reviewer: ```suggestion
Also, `ROS 2 Shapes Demo <https://eprosima-shapes-demo.readthedocs.io/en/latest/index.html>`_ is provided as a simple demo to help testing that the communication is successfully established in the system.
```

Micro-ROS provides also a set of tools for integration within many build systems, toolchains and embedded development frameworks.

.. note::

    More information about Micro-ROS can be found in :ref:`tutorials_micro_micro_tutorials`.

.. note::

    The binary package is called following the next convention ``vulcanexus-<version_name>-micro`` being the ``version_name`` the corresponding adjective associated with the :ref:`specific release <release_notes>`.
    For example, for Vulcanexus v2, ``version_name`` is ``humble``.
    Consequently, in order to install this package run the following command after setting up the repository as explained in the :ref:`installation guide <linux_binary_installation>`

        .. code-block:: bash

            sudo apt install vulcanexus-humble-micro

Vulcanexus Cloud
----------------

**Vulcanexus Cloud** is oriented to geographically spaced scenarios, deploying robotic applications through a distributed network.

> To this end, `eProsima DDS Router <https://eprosima-dds-router.readthedocs.io/en/latest/index.html>`_ is installed in this *Vulcanexus* package distribution, allowing DDS communication through different network layers (bridging the local networks through WAN communication).

Reviewer: ```suggestion
To this end, `ROS 2 DDS Router <https://eprosima-dds-router.readthedocs.io/en/latest/index.html>`_ is installed in this *Vulcanexus* package distribution, allowing ROS 2 communications through different network layers (bridging the local networks through WAN communication).
```
